vscode/src/commands/context/file-path.ts

```diff
@@ -20,13 +20,15 @@ export async function getContextFileFromUri(file: URI, range?: vscode.Range): Pr
                 return []
             }
 
+            // empty range can happen when user initiates action from right
+            // click. Treat as wanting full document.
+            range = range?.isEmpty ? undefined : range
+
             const doc = await vscode.workspace.openTextDocument(file)
             const content = doc?.getText(range).trim()
             if (!content) {
                 throw new Error('No file content')
             }
```

Comment:
```suggestion
            const endLine = Math.max(doc.lineCount - 1, 0)
            range = range ?? new vscode.Range(0, 0, endLine, 0)
            
            const isFullFile = range.end.line === doc.lineCount - 1
```

I think we still want to check the existing range to see if it's full file. If yes, then we remove it from the `return` statement below instead?

```ts
            return [
                {
                    type: 'file',
                    content,
                    uri: file,
                    source: ContextItemSource.Editor,
                    range: isFullFile? undefined : toRangeData(range),
                    size,
                },
            ] satisfies ContextItem[]
```

Reply:
aah nice catch. That probably explains some of the weirdness I was seeing.